File: frontend/src/types/JournalEntry.ts
export type JournalEntryType = 'LEARNING' | 'ACHIEVEMENT' | 'CHALLENGE'

export interface JournalEntry {
  id: string
  date: string
  type: JournalEntryType
  content: string
  createdAt: string
}

export interface CreateJournalEntryData {
  date: string
  type: JournalEntryType
  content: string
}

File: backend/src/main/java/com/squad/api/model/JournalEntry.java
package com.squad.api.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "journal_entries")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class JournalEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "uuid")
    private UUID id;

    @Column(nullable = false)
    private LocalDate date;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private JournalEntryType type;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "user_id", nullable = false)
    private String userId;
}
